scripts/nix-profile.sh.in

```diff
@@ -4,10 +4,34 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     # This part should be kept in sync with nixpkgs:nixos/modules/programs/shell.nix
 
     NIX_LINK=$HOME/.nix-profile
+    if [ -n "$XDG_DATA_HOME" ]; then
+        NIX_LINK_NEW="$XDG_DATA_HOME/nix/profile"
+    else
+        NIX_LINK_NEW=$HOME/.local/share/nix/profile
+    fi
+    if ! [ -e $NIX_LINK ]; then
+        NIX_LINK="$NIX_LINK_NEW"
+    else
+        if [ -e "$NIX_LINK_NEW" ]; then
+            warning="\033[1;35mwarning:\033[0m"
+            printf "$warning Both %s and legacy %s exist; using the latter.\n" "$NIX_LINK_NEW" "$NIX_LINK"
+            if [ "$(realpath "$NIX_LINK")" == "$(realpath "$NIX_LINK_NEW")" ]; then
+                printf "         Since the profiles match, you can safely delete either of them.\n"
+            else
+                # This should be an exceptionally rare occasion: the only way to get it would be to
+                # 1. Update to newer Nix;
+                # 2. Remove .nix-profile;
+                # 3. Set the $NIX_LINK_NEW to something other than the default user profile;
+                # 4. Roll back to older Nix.
+                # If someone did all that, they can probably figure out how to migrate the profile.
+                printf "$warning Profiles do not match. You should manually migrate from %s to %s.\n" "$NIX_LINK" "$NIX_LINK_NEW"
```

Comment:
I think this file is sourced by /etc/profile by default, am I mistaken? If so it is kind of a problem to write to stdout as it probably breaks many tools which do eval kind of things. This either needs to go to stderr and probably more safely be emitted only by interactive sessions or the nix tool.